scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java

```diff
@@ -32,1809 +32,290 @@
+ * - communicates with the Scheduler via ProActive
+ * - deals with data transfers
+ * - deals with task killing and walltime
+ * - sends result back to the Scheduler
  */
-public abstract class TaskLauncher implements InitActive {
-
-    public static final Logger logger = Logger.getLogger(TaskLauncher.class);
-
-    //Scratch dir property : we cannot take the key property from DataSpaceNodeConfigurationAgent class in RM.
-    //we should not depend from RM package in this class.
-    public static final String NODE_DATASPACE_SCRATCHDIR = "node.dataspace.scratchdir";
-
-    public static final long CLEAN_TIMEOUT = 21 * 1000; // timeout used to control max time for the cleaning operation
-
-    private boolean dataspaceInitialized = false;
-
-    public static final String EXECUTION_SUCCEED_BINDING_NAME = "success";
-    public static final String DS_SCRATCH_BINDING_NAME = "localspace";
-    public static final String DS_INPUT_BINDING_NAME = "input";
-    public static final String DS_OUTPUT_BINDING_NAME = "output";
-    public static final String DS_GLOBAL_BINDING_NAME = "global";
-    public static final String DS_USER_BINDING_NAME = "user";
-
-    public static final String MULTI_NODE_TASK_NODESET_BINDING_NAME = "nodeset";
-    public static final String MULTI_NODE_TASK_NODESURL_BINDING_NAME = "nodesurl";
-
-    /** The name used to access propagated variables map */
-    public static final String VARIABLES_BINDING_NAME = "variables";
-
-    // keep last 1024 lines ot job output by default
-    private static final int KEY_SIZE = 1024;
-
-    // to define the max line number of a task logs
-    // DO NOT USE SCHEDULER PROPERTY FOR THAT (those properties are not available on node side)
-    @Deprecated
-    public static final String OLD_MAX_LOG_SIZE_PROPERTY = "proactive.scheduler.logs.maxsize";
-    public static final String MAX_LOG_SIZE_PROPERTY = "pas.launcher.logs.maxsize";
-
-    // default log size, counted in number of log events
-    public static final int DEFAULT_LOG_MAX_SIZE = 1024;
-
-    // the prefix for log file produced in localspace
-    public static final String LOG_FILE_PREFIX = "TaskLogs";
-
-    protected DataSpacesFileObject SCRATCH = null;
-    protected DataSpacesFileObject INPUT = null;
-    protected DataSpacesFileObject OUTPUT = null;
-    protected DataSpacesFileObject GLOBAL = null;
-    protected DataSpacesFileObject USER = null;
-
-    protected NamingService namingService = null;
-    protected List<InputSelector> inputFiles;
-    protected List<OutputSelector> outputFiles;
+@ActiveObject
+public class TaskLauncher {
 
-    // buffered string to store logs destined to the client (for e.g. datapspaces error/warn messages)
-    private StringBuffer clientLogs;
+    private static final Logger logger = Logger.getLogger(TaskLauncher.class);
 
-    protected Decrypter decrypter = null;
+    private ProActiveForkedTaskLauncherFactory factory = new ProActiveForkedTaskLauncherFactory();
```

Comment:
Is it required to initialize it for all TaskLauncher instances? it seems that the factory should be passed in the constructor?